File: 351002/Khmel/Task320/Repositories/WriterRepository.cs
public class WriterRepository : IRepository<Writer>
{
    private readonly AppDbContext _context;

    public WriterRepository(AppDbContext context)
    {
        _context = context;
    }

    public Writer? GetById(long id)
    {
        return _context.Writers.Find(id);
    }

    public IEnumerable<Writer> GetAll()
    {
        return _context.Writers.ToList();
    }

    public Writer Create(Writer entity)
    {
        _context.Writers.Add(entity);
        _context.SaveChanges(); 
        return entity;
    }

    public Writer Update(Writer entity)
    {
        var existing = _context.Writers.Find(entity.Id);
        if (existing == null)
            throw new KeyNotFoundException($"Writer с id={entity.Id} не найден");

        existing.Login = entity.Login;
        existing.Password = entity.Password;
        existing.Firstname = entity.Firstname;
        existing.Lastname = entity.Lastname;

        _context.SaveChanges(); 
        return existing;
    }

    public bool Delete(long id)
    {
        var entity = _context.Writers.Find(id);
        if (entity == null) return false;

        _context.Writers.Remove(entity);
        _context.SaveChanges();
        return true;
    }
}

File: 351003/Guzaev/publisher/app/repositories/writer_repository.py
from sqlalchemy.orm import Session
from database import SessionLocal
from models.models import Writer

class WriterRepository:
    def __init__(self):
        self.db: Session = SessionLocal()

    def create(self, login: str, password: str, firstname: str, lastname: str, role: str = "CUSTOMER") -> Writer:
        writer = Writer(
            login=login,
            password=password,
            firstname=firstname,
            lastname=lastname,
            role=role
        )
        self.db.add(writer)
        self.db.commit()
        self.db.refresh(writer)
        return writer

    def get_all(self):
        return self.db.query(Writer).all()

    def get_by_id(self, writer_id: int):
        return self.db.query(Writer).filter(Writer.id == writer_id).first()

    def get_by_login(self, login: str):
        return self.db.query(Writer).filter(Writer.login == login).first()

    def update(self, writer_id: int, **kwargs) -> Writer:
        writer = self.get_by_id(writer_id)
        if not writer:
            return None
        for key, value in kwargs.items():
            if hasattr(writer, key) and value is not None:
                setattr(writer, key, value)
        self.db.commit()
        self.db.refresh(writer)
        return writer

    def delete(self, writer_id: int) -> bool:
        writer = self.get_by_id(writer_id)
        if not writer:
            return False
        self.db.delete(writer)
        self.db.commit()
        return True